docs/HTTP-batchsource.md

```diff
@@ -0,0 +1,206 @@
+
+**CSV Skip First Row:** If true, skip first row. This should be set in case first row of csv is a header row.
+
+### Basic Authentication
+
+**Username:** Username for basic authentication.
+
+**Password:** Password for basic authentication.
+
+### HTTP Proxy
+
+**Proxy URL:** Proxy URL. Must contain a protocol, address and port.
+
+**Username:** Proxy username.
+
+**Password:** Proxy password.
+
+### Error Handling
+
+**HTTP Errors Handling:** A mapping which defines error handling strategy used, when pages return certain HTTP status
+codes. The left column contains a regular expression for http status code. The right column contains an action which
+is done in case of match. If http status code matches multiple regular expressions, the first specified in mapping
+is matched.
+
+Example:
+
+| HTTP Code Regexp  | Error Handling          |
+| ----------------- |:-----------------------:|
+| 2..               | Success                 |
+| 401               | Retry and fail          |
+| 4..               | Fail                    |
+| 5..               | Retry and send to error |
+| .*                | Fail                    |
+
+Note: pagination types "Link in response header", "Link in response body", "Token in response body" does not support
+"Send to error", "Skip", "Retry and send to error", "Retry and skip" options.
+
+**Non-HTTP Error Handling:** Strategy used to handle errors during transformation of a text entry to record.
+Possible values are:<br>
+Stop on error - Fails pipeline due to erroneous record.<br>
+Send to error - Sends erroneous record's text to error port and continues.<br>
+Skip on error - Ignores erroneous records.
+
+**Retry Policy:** Policy used to calculate delay between retries.
+
+**Linear Retry Interval:** Interval between retries. Is only used if retry policy is "linear".
+
+**Max Retry Duration:** Maximum time in seconds retries can take.
+
+**Connect Timeout:** Maximum time in seconds connection initialization is allowed to take.
+
+**Read Timeout:** Maximum time in seconds fetching data from the server is allowed to take.
+
+### Pagination
+
+**Pagination Type:** Strategy used to determine how to get next page.
+
+**Wait time between pages:** Interval between loading pages in milliseconds.
+<br><br>
+
+##### Pagination type: None
+Only single page is loaded.
+<br><br>
+
+##### Pagination type: Link in response header
+In response there is a "Link" header, which contains an url marked as "next". Example:<br>
+`<http://example.cdap.io/admin/api/pages?page=1&q.language.id=1>; rel="first",
+ <http://example.cdap.io/admin/api/pages?page=2&q.language.id=1>; rel="next",
+ <http://example.cdap.io/admin/api/pages?page=2&q.language.id=1>; rel="last"`
+<br><br>
+
+##### Pagination type: Link in response body
+Every page contains a next page url. This pagination type is only supported for JSON and XML formats.
+Pagination happens until no next page field is present or until page contains no elements.
```

Comment:
Should include an example to make it more clear.

Reply:
done